lib/auth.ts

```diff
@@ -133,6 +147,81 @@ export class Auth {
     return token;
   };
 
+  /**
+   * Perform a request against Kontist REST API
+   */
+  private request = async (path: string, method: HttpMethod, body?: string) => {
+    if (!this.token) {
+      throw new Error("User unauthorized");
+    }
+
+    const requestUrl = new URL(path, this.baseUrl).href;
+
+    const response = await fetch(requestUrl, {
+      method,
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.token.accessToken}`
+      },
+      body
+    });
+
+    return response.json();
+  };
+
+  /**
+   * Called periodically by `createMFAChallenge` until
+   * the challenge expires or its status is updated
+   */
+  private pollChallengeStatus = (
+    pendingChallenge: Challenge,
+    resolve: Function,
+    reject: Function
+  ) => async () => {
+    const challenge = await this.request(
+      `${MFA_CHALLENGE_PATH}/${pendingChallenge.id}`,
+      HttpMethod.GET
+    );
+
+    if (new Date(challenge.expiresAt) < new Date()) {
+      if (this.challengePollIntervalId) {
+        clearInterval(this.challengePollIntervalId);
+      }
+      reject(new Error("Challenge expired"));
```

Comment:
can you add explicit `return` to those reject/resolve calls? 🙏🏻

Reply:
Sure 🙂 added here: e9ecb7de0e6e2fe3a4187f87602dcbc8ef575e9f